pages/spacedeck/[[...roomId]].js

```diff
@@ -0,0 +1,297 @@
+                    await createSketchRoom(link, sketch.name, parent);
+                }
+            };
+            const syncSketches = await spacedeck.syncAllSketches()
+                .catch((error) => {
+                    console.debug(error);
+                    setIsSketchServerDown(true);
+                });
+            syncSketches && await updateStructure(spacedeck.getStructure());
+            setSyncingServerSketches(false);
+        };
+
+        if (!cancelled && serviceSpaceId && serverSketches) {
+            syncServerSketchesWithMatrix();
+        }
+
+        return () => cancelled = true;
+        // eslint-disable-next-line react-hooks/exhaustive-deps
+    }, [serviceSpaceId, serverSketches]);
+
+    useEffect(() => {
+        let cancelled = false;
+        const populateSketchesfromServer = async (recursion) => {
+            if (!isEmpty(spacedeck.getStructure())) {
+                setServerSketches(spacedeck.getStructure());
+            } else if (!recursion) {
+                await spacedeck.syncAllSketches();
+                populateSketchesfromServer(true);
+            }
+        };
+        !cancelled && getConfig().publicRuntimeConfig.authProviders.spacedeck.baseUrl && populateSketchesfromServer();
+
+        return () => {
+            cancelled = true;
+        };
+    }, [spacedeck]);
+
+    useEffect(() => {
+        let cancelled = false;
+
+        !cancelled && setContent(matrix.roomContents.get(roomId));
+
+        return () => cancelled = true;
+    }, [matrix.roomContents, roomId]);
+
+    async function createSketchRoom(link, name, parent = serviceSpaceId) {
+        // eslint-disable-next-line no-undef
+        if (process.env.NODE_ENV === 'development') console.debug('creating room for ' + name);
+        const room = await matrix.createRoom(name, false, '', 'invite', 'content', 'sketch-link').catch(() => {
+            setErrorMessage(t('Something went wrong when trying to create a new room'));
+        });
+        await auth.getAuthenticationProvider('matrix').addSpaceChild(parent, room).catch(() => {
+            setErrorMessage(t('Couldn\'t add the new room to your sketch folder'));
+        });
+        await matrixClient.sendMessage(room, {
+            msgtype: 'm.text',
+            body: link,
+        }).catch(() => {
+            setErrorMessage(t('Something went wrong when trying to save the new sketch link'));
+        });
+
+        return room;
+    }
+
+    const removeLink = async () => {
+        setRemovingLink(true);
+        const remove = await spacedeck.deleteSpaceById(content.body.substring(content.body.lastIndexOf('/') + 1)).catch((e) => console.log(e));
+        if (!remove) {
+            setRemovingLink(false);
+
+            return;
+        }
+        if (!remove.ok) {
+        // @TODO callback function to give user feedback when removing on the server fails
```

Comment:
Regarding this `@TODO`: I know some people are not the biggest fan of `alert()`, **but** in lack of better solutions how about telling the user that deleting the link failed and letting them know that the /sketch deck -still- exists?

I guess this is an edge case and in most cases deleting will work just fine anyways.